packages/next/src/lib/typescript/writeAppTypeDeclarations.ts

```diff
@@ -49,16 +49,14 @@ export async function writeAppTypeDeclarations({
     directives.push('/// <reference types="next/image-types/global" />')
   }
 
-  if (isAppDirEnabled) {
-    if (hasPagesDir) {
-      directives.push(
-        '/// <reference types="next/navigation-types/compat/navigation" />'
-      )
-    } else {
-      directives.push(
-        '/// <reference types="next/navigation-types/navigation" />'
-      )
-    }
+  if (isAppDirEnabled && hasPagesDir) {
+    directives.push(
+      '/// <reference types="next/navigation-types/compat/navigation" />'
+    )
+  } else {
+    directives.push(
+      '/// <reference types="next/navigation-types/navigation" />'
+    )
```

Comment:
No. It means that `/// <reference types="next/navigation-types/navigation" />` will be added if appDir is not enabled. It is not the intended behavior.